kpops/cli/main.py

```diff
@@ -130,37 +144,51 @@ def get_step_names(steps_to_apply: list[PipelineComponent]) -> list[str]:
     return [step.name.removeprefix(step.prefix) for step in steps_to_apply]
 
 
-def filter_steps_to_apply(
-    pipeline: Pipeline, steps: set[str]
+def filter_component(
+    pipeline: Pipeline, steps: set[str], filter_type: FilterType
 ) -> list[PipelineComponent]:
-    skipped_steps: list[str] = []
+    filtered_steps = []
+    if filter_type == FilterType.include:
+        filtered_steps = list(
+            filter(
+                lambda component: component.name.removeprefix(component.prefix)
+                in steps,
+                pipeline,
+            )
+        )
+        log.info(f"Including the following steps: {get_step_names(filtered_steps)}")
+    elif filter_type == FilterType.exclude:
+        filtered_steps = list(
+            filter(
+                lambda component: component.name.removeprefix(component.prefix)
+                not in steps,
+                pipeline,
+            )
+        )
+        log.info(f"Excluding the following steps: {get_step_names(filtered_steps)}")
+    return filtered_steps
 
-    def filter_component(component: PipelineComponent) -> bool:
-        step_name = component.name.removeprefix(component.prefix)
-        if step_name in steps:
-            return True
-        skipped_steps.append(step_name)
-        return False
 
-    steps_to_apply = list(filter(filter_component, pipeline))
+def filter_steps_to_apply(
+    pipeline: Pipeline, steps: set[str], filter_type: FilterType
+) -> list[PipelineComponent]:
     log.info("KPOPS_PIPELINE_STEPS is defined.")
-    log.info(
-        f"Executing only on the following steps: {get_step_names(steps_to_apply)}"
-        f", \n ignoring {skipped_steps}"
-    )
+    steps_to_apply = filter_component(pipeline, steps, filter_type)
     return steps_to_apply
```

Comment:
what's the purpose of this function if it only calls the other one?

Reply:
True, I refactored it